modules/core/src/main/java/org/apache/ignite/internal/managers/deployment/GridDeploymentLocalStore.java

```diff
@@ -183,40 +183,43 @@
                 return null;
             }
 
-            try {
-                // Check that class can be loaded.
-                String clsName = meta.className();
+            while (true) {
```

Comment:
Is it safe to loop forever like this? Maybe add some limit counter to the loop?

Reply:
I agree to add a limit, but when we can't find deployment here node will only have to stop by FH.